lib/logstash/filters/csv.rb

```diff
@@ -91,5 +98,44 @@ def filter(event)
 
   end # def filter
 
+  private
+
+  def ignore_field?(index)
+    !@columns[index] && !@autogenerate_column_names
+  end
+
+  def should_convert?(field_name)
+    !@convert[field_name].nil?
+  end
+
+  def convert(field_name, value)
+    transformation = @convert[field_name].to_sym
+    converters[transformation].call(value)
+  end
+
+  def converters
+    @converters ||= {
+      :integer => lambda do |value|
+        CSV::Converters[:integer].call(value)
+      end,
+      :float => lambda do |value|
+        CSV::Converters[:float].call(value)
+
+      end,
+      :date => lambda do |value|
+        CSV::Converters[:date].call(value)
+
+      end,
+      :date_time => lambda do |value|
+        CSV::Converters[:date_time].call(value)
```

Comment:
As the convert hash is not validated, consider using fetch and consider using the Converters[:symbol] as a callable directly, e.g.
```ruby
@converters ||= begin
  hash = Hash.new
  [:integer, :float, :date, :date_time].each do |sym|
    hash[sym] = CSV::Converters[sym]
  end
  hash[:boolean] = ...
  hash
end
```
You can probably create the `@converters` ivar once in `register`.
Using fetch in convert method, e.g. if the transformation was spelled incorrectly "bollean"
```ruby
conv = converters.fetch(transformation) do |key|
  @logger.error("No converter found for transform", :transformation => key)
  DoNothingConverter.new
end 
conv.call(value)
```